guardian/cdk
# feat: Add "new" command for cdk creation (ported from cdk-cli repo)

## What does this change?


This PR follows on from https://github.com/guardian/cdk/pull/962
It adds in the actual functionality to generate a cdk project. This is almost identical to what was in the cdk-cli repository except for the fact we are using yargs instead of oclif for parsing the input arguments.

A future pr will refactor this to optimise the code and remove any unused parts.

## How to test



From the cdk directory run ./script/cli new --app my-app --stack my-stack and confirm cdk directory is created successfully

## How can we measure success?

The "new" script successfully creates a cdk folder and proper elements when run.

## Have we considered potential risks?
 minimal risk. This is porting over existing functionality to a new repo. The change is for stand-alone functionality. No impact on other things.

## Checklist

- [x] I have listed any breaking changes, along with a migration path.[^1] _There are no breaking changes in this PR._

[^1]: Consider whether this is something that will mean changes to projects that have already been migrated, or to the CDK CLI tool. If changes are required, consider adding a checklist here and/or linking to related PRs.
[^2]: If you are adding a new construct or pattern, has new documentation been added? If you are amending defaults or changing behaviour, are the existing docs still valid?
